gradoop-flink/src/main/java/org/gradoop/flink/util/FlinkAsciiGraphLoader.java

```diff
@@ -136,17 +136,15 @@ public LogicalGraph getLogicalGraph() {
    * @return logical graph of vertex and edge space
    */
   public LogicalGraph getLogicalGraph(boolean withGraphContainment) {
-    GraphHead graphHead = config.getGraphHeadFactory()
-      .createGraphHead(GradoopConstants.DB_GRAPH_LABEL);
-
     if (withGraphContainment) {
-      ExecutionEnvironment env = config.getExecutionEnvironment();
-
-      return config.getLogicalGraphFactory().fromDataSets(env.fromElements(graphHead),
-        env.fromCollection(getVertices()).map(new AddToGraph<>(graphHead)),
-        env.fromCollection(getEdges()).map(new AddToGraph<>(graphHead)));
-
+      return config.getLogicalGraphFactory().fromCollections(getVertices(), getEdges())
+        .transformGraphHead((head, transformed) -> {
```

Comment:
I recently had some problems with lambdas during remote execution or shell based execution. Did you tested this one outside your IDE?

Reply:
Optional approach:
```
private GraphHead setDefaultLabel(GraphHead current, GraphHead transformed) {
    current.setLabel(GradoopConstants.DB_GRAPH_LABEL);
    return current;
  }
```
An call it:
```
.transformGraphHead(this::setDefaultLabel);
```